oci/layout/oci_dest.go

```diff
@@ -184,23 +193,20 @@ func (d *ociImageDestination) PutManifest(m []byte) error {
 		Architecture: runtime.GOARCH,
 		OS:           runtime.GOOS,
 	}
-	d.index.Manifests = append(d.index.Manifests, desc)
+	d.addManifest(&desc)
 
 	return nil
 }
 
-func ensureDirectoryExists(path string) error {
-	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
-		if err := os.MkdirAll(path, 0755); err != nil {
-			return err
+func (d *ociImageDestination) addManifest(desc *imgspecv1.Descriptor){
+	for i, manifest := range d.index.Manifests {
+		if manifest.Digest.String() == desc.Digest.String() {
```

Comment:
I think #318 is not doing the right thing.

> But how do I then reference multiple images within the same OCI directory?

Use different tags, I'm not sure I understand what you mean.

> With path I mean the directory of where the OCI index lives.

There's only one OCI index at the root `index.json` (relative to the path where the OCI layout exists). I'd recommend taking a look at https://github.com/openSUSE/umoci's interface to see what I mean (disclosure, I wrote it).